decentraland/unity-explorer
# fix: Missing modals and unblocked inputs in Reward-Claim scene

## What does this PR change?
Fix #1442 
Fix #1443 

This PR includes 2 fixes:
- **The correct blocking/unblocking of the player input when we write in any UIInput SDK component**: This avoid to break the flow when we press keys like [E], [B] or others while we're writting inside the input textbox.
- **The correct management of the responses from `signedFetch` / `simpleFetch`**: Before this change, all the bad requests generated by a `signedFetch` / `simpleFetch` were not being sent back to the scene so it was not able to manage the error cases correctly (in the case of the Reward-Claim scene, the UI informing about the error when we introduced a wrong CAPTCHA was not appearing).

## How to test the changes?
1. Launch the explorer.
2. Go to Goerli and go to the coords `82,-6`.
3. Claim the reward.
4. When the CAPTCHA window appears, write something wrong.
![image](https://github.com/user-attachments/assets/1ad95d57-ab61-4f5e-bec5-cd11dab94527)
5. Notice that this window closes and another one appears informing about the error.
![image](https://github.com/user-attachments/assets/3b3d0e29-1fdd-4559-8a6a-4508b50b58a0)
6. Repeat the step 3.
7. When the CAPTCHA window appears, write the correct characters.
![image](https://github.com/user-attachments/assets/2308f068-8f7d-4a68-88bf-dcfd930802c7)
8. Notice that this window closes and another one appears informing about the wearable is coming successfully.
![image](https://github.com/user-attachments/assets/b94ad799-f36b-492e-bb4a-cd3912be7d20)
9. Also check, in the steps 4 and 7, that we can press any key while we have the focus in the input textbox whitout any conflict.

**NOTES**: It's normal that the CAPTCHA UI and other modals appear with a weird format, this is because the scene is not building them correctly and is also happening in prod 👍 

![image](https://github.com/user-attachments/assets/b143f604-613a-4c0f-8244-8e4d38e23343)

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
